src/components/Map/MapView.tsx

```diff
@@ -32,6 +45,30 @@ export const MapView = () => {
   const activeNodeId = useSelector(selectActiveNodeId());
   const { edgesFeatureCollection, viewState } = useSelector(selectMapState());
 
+  const waypoints = useSelector(selectAllWaypoints());
+  const isWaypointEdit = useSelector(selectIsWaypointEdit());
+  const newWaypoint = useSelector(selectIsNewWaypoint());
+
+  const handleClick = (e: MapLayerMouseEvent) => {
+    // Can only create new waypoint if the state is toggled
+    if (newWaypoint) {
+      const createdWaypoint: Waypoint = {
+        id: 0,
+        latitudeI: Math.round(e.lngLat.lat * 1e7), // Location clicked
+        longitudeI: Math.round(e.lngLat.lng * 1e7),
+        expire: Math.round(moment().add(1, "years").valueOf() / 1000), // Expires one year from today
+        lockedTo: 0, // Locked to me
```

Comment:
0 signifies an open waypoint, not a locked waypoint. In this case, we would set the `waypoint.lockedTo` field to our `nodeId`. This being said, I would imagine open waypoints would be a good default

https://buf.build/meshtastic/protobufs/docs/main:meshtastic#meshtastic.Waypoint